src/visualization/plot_recipes.jl

```diff
@@ -278,20 +278,179 @@ end
+
+  center = mesh.tree.center_level_0[1]
+  domain_length = mesh.tree.length_level_0
+  x = cache.elements.node_coordinates
+
+  variable_names = SVector(varnames(solution_variables, equations))
+
+  return PlotData1D(vec(x), u, variable_names, vcat(x[1, 1, :], x[1, end, end]))
+end
+
+PlotData1D(u_ode::AbstractVector, semi; kwargs...) = PlotData1D(wrap_array(u_ode, semi), semi; kwargs...)
+
+#Plot from a solution object, which is directly return when running a simulation.
+PlotData1D(sol::TrixiODESolution; kwargs...) = PlotData1D(sol.u[end], sol.prob.p; kwargs...)
+
+struct PlotMesh1D{PD<:PlotData1D}
+  plot_data::PD
+end
+
+getmesh(pd::PlotData1D) = PlotMesh1D(pd)
+
+@recipe function f(pd::PlotData1D)
+  label --> pd.variable_names[1]
+  pd.x, pd.data
+end
+
+@recipe function f(pm::PlotMesh1D)
+  @unpack plot_data = pm
+  @unpack x, mesh_vertices_x = plot_data
+
+  # Set geometric and annotation properties
+  xlims --> (x[begin], x[end])
+  legend -->  :none
+
+  # Set series properties
+  seriestype --> :vline
+  linecolor --> :grey
+  linewidth --> 1
+
+  # Return data for plotting
+  mesh_vertices_x
+end
+
+#Store multiple PlotData1D objects in one PlotDataSeries1D.
+#This is used for multi-variable equations.
+struct PlotDataSeries1D{PD<:PlotData1D}
+  plot_data::PD
+  variable_id::Int
+end
+
+Base.firstindex(pd::PlotData1D) = first(pd.variable_names)
+Base.lastindex(pd::PlotData1D) = last(pd.variable_names)
+Base.length(pd::PlotData1D) = length(pd.variable_names)
+Base.size(pd::PlotData1D) = (length(pd),)
+Base.keys(pd::PlotData1D) = tuple(pd.variable_names...)
+Base.eltype(pd::PlotData1D) = Pair{String, PlotDataSeries1D}
+function Base.iterate(pd::PlotData1D, state=1)
+  if state > length(pd)
+    return nothing
+  else
+    return (pd.variable_names[state] => pd[pd.variable_names[state]], state + 1)
+  end
 end
 
+function Base.show(io::IO, pd::PlotData1D)
+  print(io, "PlotData1D{",
+            typeof(pd.x), ",",
+            typeof(pd.data), ",",
+            typeof(pd.variable_names), ",",
+            typeof(pd.mesh_vertices_x),
+            "}(<x>, <data>, <variable_names>, <mesh_vertices_x>)")
+end
+
+function Base.getindex(pd::PlotData1D, variable_name)
+  variable_id = findfirst(isequal(variable_name), pd.variable_names)
+
+  if isnothing(variable_id)
+    throw(KeyError(variable_name))
+  end
+
+  return PlotDataSeries1D(pd, variable_id)
+end
+
+function Base.show(io::IO, pm::PlotMesh1D)
+  print(io, "PlotMesh1D{", typeof(pm.plot_data), "}(<plot_data::PlotData1D>)")
+end
+
+#Create plot from a PlotDataSeries1D object and creates subplots for each variable.
+@recipe function f(pds::PlotDataSeries1D)
+  @unpack plot_data, variable_id = pds
+  @unpack x, data, variable_names = plot_data
+
+  # Set geometric properties
+  xlims --> (x[begin], x[end])
+
+  # Set annotation properties
+  legend -->  :none
+  title --> variable_names[variable_id]
+  #colorbar --> :true
+
+  # Return data for plotting
+  x, vec(data[variable_id,:,:])
```

Comment:
I don't like it, that we do some conversion from a solver data representation to a plot data representation here. Like in 2D, all this should already be handled in the "main constructor" for `PlotData1D`, i.e., the one that accepts `u` and `semi`. In here there should only be something like `data[variable_id]` to select the correct data set.